### PR TITLE
release ENIs

### PR DESCRIPTION
**Pull request by kshychko**
per #251

### CODE REVIEW

File: aws/utility.sh [hunk 1]
```diff
@@ -1276,3 +1276,19 @@ function invalidate_distribution() {
...
+    local attachment_id
+    local network_interface_id
+    local eni_list_file="$( getTempFile eni_list_XXXX.json)"
+    
+    aws --region "${region}" ec2 describe-network-interfaces --filters Name=requester-id,Values=*${requester_id} > "${eni_list_file}" || return $?
+    
+    attachment_id=$(jq -r ".NetworkInterfaces[].Attachment.AttachmentId" < "${eni_list_file}") || return $?
+    network_interface_id=$(jq -r ".NetworkInterfaces[].NetworkInterfaceId" < "${eni_list_file}") || return $?
+    aws --region "${region}" ec2 detach-network-interface --attachment-id "${attachment_id}" || return $?
+    aws --region "${region}" ec2 delete-network-interface --network-interface-id "${attachment_id}" || return $?
```
Comment (roleyfoley) [inline aws/utility.sh]: Should 
`
 --network-interface-id "${attachment_id}"  actually be --network-interface-id "${network_interface_id}" ?

Comment (kshychko) [reply aws/utility.sh]: that's correct! let me fix it

File: aws/templates/application/application_lambda.ftl [hunk 2]
```diff
@@ -306,6 +306,27 @@
                                 )
                             ) /]
                 [/#if]
+                [@cfScript
```
Comment (roleyfoley) [inline aws/templates/application/application_lambda.ftl]: Would we need to check that the lambda is attached to a VPC? Will the command fail if there are no interfaces found?

Comment (kshychko) [reply aws/templates/application/application_lambda.ftl]: @roleyfoley checks have been added

**Comment by kshychko** [comment]
@ml019 the loop has been added, please review

File: aws/utility.sh [hunk 3]
```diff
@@ -1276,3 +1276,23 @@ function invalidate_distribution() {
...
+# -- ENI interface removal  --
+function release_enis() {
+    local region="$1"; shift
+    local requester_id="$1"; shift
+    local eni_list_file="$( getTempFile eni_list_XXXX.json)"
+    
+    aws --region "${region}" ec2 describe-network-interfaces --filters Name=requester-id,Values=*${requester_id} > "${eni_list_file}" || return $?
+    
+    for attachment_id in $( jq -r '.NetworkInterfaces[].Attachment.AttachmentId' < "${eni_list_file}" ) ; do
+        if [[ -z "${attachment_id}" ]] && [[ -n "${attachment_id}" ]]; then
```
Comment (ml019) [inline aws/utility.sh]: Don't think this will ever work - -z and -n are opposites

File: aws/utility.sh [hunk 4]
```diff
@@ -1276,3 +1276,23 @@ function invalidate_distribution() {
...
+            aws --region "${region}" ec2 detach-network-interface --attachment-id "${attachment_id}" || return $?
+        fi
+    done
+    for network_interface_id in $( jq -r '.NetworkInterfaces[].NetworkInterfaceId' < "${eni_list_file}" ) ; do
+        if [[ -z "${network_interface_id}" ]] && [[ -n "${network_interface_id}" ]]; then
```
Comment (ml019) [inline aws/utility.sh]: see previous comment

**Comment by kshychko** [comment]
@ml019 if conditions have been fixed

**Pull request merged**